### DISCUSSION TITLE
Feature Request: Centered code mode (like intellij's distraction free mode)

### DISCUSSION THREAD

**Discussion by ldubos** [helix-editor/helix#10357] [Ideas]
Hi, I find Intellij's "distaction free mode" really interesting, especially the fact of centering the code on the screen. I find it much less tiring than having to constantly look at the left side of the screen, which happens quite often with helix when you have only one window opened, because there is no sidebar which in an editor like VS Code center the code even if it's not the intention at the beginning.

So it could be cool to having an option in config file to allow the user to choose to center his code in the window, maybe with exception like when we are in split-screen.

an example of the feature (in vscode):
![image](https://user-images.githubusercontent.com/16986491/181006749-976412c7-a1f3-43ff-bd56-6e218ee551b3.png)

**Comment by AceofSpades5757** [comment]
I think this usually gets implemented as a plugin, like with goyo.vim for Vim. Different view options, or more control over the view would be nice.

**Comment by archseer** [comment]
Seems like it overlaps https://github.com/helix-editor/helix/issues/2262

**Comment by sicher** [comment]
For now you can do `:set gutters.line-numbers.min-width 30` or whatever matches your terminal width, text-width and soft wrap settings.

**Comment by gavales** [comment]
> For now you can do `:set gutters.line-numbers.min-width 30` or whatever matches your terminal width, text-width and soft wrap settings.

TBH this will more than satisfy me for now, the way it behaves is what I desire, especially when combined with `soft-wrap.wrap-at-text-width`.

When a scripting language is implemented, you could easily write a plugin to calculate the correct `gutters.line-numbers.min-width` to centre the viewport. As long as the maximum value `gutters.line-numbers.min-width` doesn't get restricted like it is in vim, that is. :)

**Comment by shubhamkumar13** [comment]
is this possible now?

I am able to only use `:set gutters.line-numbers.min-width 40` to get this view

<img width="1920" height="1140" alt="image" src="https://github.com/user-attachments/assets/048fc027-9131-4506-9069-4c95adfb5973" />

**Comment by bcspragu** [comment]
I also wanted something like this, so I built a simple wrapper that does it: https://git.sr.ht/~bsprague/hx-zen-config

TL;DR - It reads your Helix config, adds the settings to make it "distraction free/zen", writes that to a temp file, then runs Helix with the `--config` flag. It probably breaks under certain circumstances (e.g. if you have multiple active Helix configs, like `/etc/helix/...` and `~/.config/helix/...`), but it works great for my purposes.

**Comment by shubhamkumar13** [comment]
Thanks @bcspragu  <3

I'll try and let you know
